core/vtk/ttkMergeTreeAutoencoder/ttkMergeTreeAutoencoder.h

```diff
@@ -0,0 +1,591 @@
+  }
+
+  void SetEpsilonTree1(double epsilonTree1) {
+    epsilonTree1_ = epsilonTree1;
+    oldEpsilonTree1 = epsilonTree1_;
+    Modified();
+    resetDataVisualization();
+  }
+  double GetEpsilonTree1() {
+    return epsilonTree1_;
+  }
+
+  void SetEpsilon2Tree1(double epsilon2Tree1) {
+    epsilon2Tree1_ = epsilon2Tree1;
+    Modified();
+    resetDataVisualization();
+  }
+  double GetEpsilon2Tree1() {
+    return epsilon2Tree1_;
+  }
+
+  void SetEpsilon3Tree1(double epsilon3Tree1) {
+    epsilon3Tree1_ = epsilon3Tree1;
+    Modified();
+    resetDataVisualization();
+  }
+  double GetEpsilon3Tree1() {
+    return epsilon3Tree1_;
+  }
+
+  void SetPersistenceThreshold(double persistenceThreshold) {
+    persistenceThreshold_ = persistenceThreshold;
+    Modified();
+    resetDataVisualization();
+  }
+  double GetPersistenceThreshold() {
+    return persistenceThreshold_;
+  }
+
+  void SetDeleteMultiPersPairs(bool delMultiPersPairs) {
+    deleteMultiPersPairs_ = delMultiPersPairs;
+    Modified();
+    resetDataVisualization();
+  }
+  bool GetDeleteMultiPersPairs() {
+    return deleteMultiPersPairs_;
+  }
+
+  // Testing options
+  void SetNodePerTask(int nodePerTask) {
+    nodePerTask_ = nodePerTask;
+    Modified();
+    resetDataVisualization();
+  }
+  int GetNodePerTask() {
+    return nodePerTask_;
+  }
+
+  // Output options
+  void SetCreateOutput(bool createOutput) {
+    createOutput_ = createOutput;
+    Modified();
+    resetDataVisualization();
+  }
+  bool GetCreateOutput() {
+    return createOutput_;
+  }
+
+  /**
+   * This static method and the macro below are VTK conventions on how to
+   * instantiate VTK objects. You don't have to modify this.
+   */
+  static ttkMergeTreeAutoencoder *New();
+  vtkTypeMacro(ttkMergeTreeAutoencoder, ttkAlgorithm);
+
+protected:
+  /**
+   * Implement the filter constructor and destructor
+   * (see cpp file)
+   */
+  ttkMergeTreeAutoencoder();
+
+  /**
+   * Specify the input data type of each input port
+   * (see cpp file)
+   */
+  int FillInputPortInformation(int port, vtkInformation *info) override;
+
+  /**
+   * Specify the data object type of each output port
+   * (see cpp file)
+   */
+  int FillOutputPortInformation(int port, vtkInformation *info) override;
+
+  /**
+   * Pass VTK data to the base code and convert base code output to VTK
+   * (see cpp file)
+   */
+  int RequestData(vtkInformation *request,
+                  vtkInformationVector **inputVector,
+                  vtkInformationVector *outputVector) override;
+
+  int run(vtkInformationVector *outputVector,
+          std::vector<vtkSmartPointer<vtkMultiBlockDataSet>> &inputTrees,
+          std::vector<vtkSmartPointer<vtkMultiBlockDataSet>> &inputTrees2);
+
+  int runCompute(
+    vtkInformationVector *outputVector,
+    std::vector<vtkSmartPointer<vtkMultiBlockDataSet>> &inputTrees,
+    std::vector<vtkSmartPointer<vtkMultiBlockDataSet>> &inputTrees2);
+
+  void makeOneOutput(
```

Comment:
why is that function declared here?
it seems it is also defined in the utils class. (same remark for the other functions below)

Reply:
Removed in the next commit!

Reply:
Thanks!